src/client/datascience/notebook/defaultCellLanguageService.ts

```diff
@@ -38,7 +40,11 @@ export class NotebookCellLanguageService implements IExtensionSingleActivationSe
             metadata?.language_info?.name ||
             (metadata?.kernelspec as IJupyterKernelSpec | undefined)?.language ||
             this.lastSavedNotebookCellLanguage;
-        return translateKernelLanguageToMonaco(jupyterLanguage || PYTHON_LANGUAGE);
+
+        // Default to python language only if the Python extension is installed.
+        const defaultLanguage = this.pythonExtensionChecker.isPythonExtensionInstalled ? PYTHON_LANGUAGE : 'plaintext';
```

Comment:
I think this wasn't right.

Reply:
If user doesn't have a python kernel at all, no point defaulting to Python

Reply:
Else later if you save these notebooks and open it, then we assume its a python notebook and ask the user to install Python.